static_frame/core/index_hierarchy.py

```diff
@@ -1649,15 +1684,15 @@ def rehierarch(self: IH,
             own_blocks=True,
             )
 
-    def _get_unique_labels_in_occurence_order(self: IH,
+    def get_unique_labels_in_occurence_order(self: IH,
```

Comment:
I was going to suggest naming this `unique_at_depth` but then I noticed the signature of `unique` already includes a `depth` parameters. Do we need this new interface, or can we extend `unique()` to do what this does (if they are different).

Reply:
Maybe we need to add  a Boolean `order_by_occurrence` flag to `unique()` on this and `Index`; that way users can opt-in to getting occurrence order.

Reply:
I really, really like this idea!! I never liked the verbosity of `get_unique_labels_in_occurence_order`, and yet I wasn't able to come up with any ideas.